src/main/java/com/iwaconsolti/league/manager/controller/LeaguesController.java

```diff
@@ -0,0 +1,129 @@
+package com.iwaconsolti.league.manager.controller;
+
+import com.iwaconsolti.league.manager.model.Matches;
+import com.iwaconsolti.league.manager.model.Players;
+import com.iwaconsolti.league.manager.model.Teams;
+import com.iwaconsolti.league.manager.service.ILeagues;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.context.annotation.Profile;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Profile("default")
+@RestController
+@RequestMapping(value = "/leagues/{leagueType}")
+@Slf4j
+public class LeaguesController {
+
+    private ILeagues soccerLeagues;
+    private ILeagues baseBallLeague;
+
+    public LeaguesController(@Qualifier("soccerLeague") ILeagues soccerLeague, @Qualifier("baseBallLeague") ILeagues baseBallLeague) {
+        this.soccerLeagues = soccerLeague;
+        this.baseBallLeague = baseBallLeague;
+        log.info("Default LeaguesController created");
+    }
+
+    @PostMapping("/team")
+    public ResponseEntity<Teams> createTeam(@PathVariable String leagueType, @RequestBody Teams newTeam) {
+        if (leagueType.equals("soccer")) {
+            Teams teamSoccer = soccerLeagues.saveTeams(newTeam);
+            log.info("Team created");
+            return ResponseEntity.ok(teamSoccer);
+        } else {
+            Teams teamBase = baseBallLeague.saveTeams(newTeam);
+            return ResponseEntity.ok(teamBase);
+        }
+    }
+
+    @PostMapping("/player")
+    public ResponseEntity<Players> createPlayer(@PathVariable String leagueType, @RequestBody Players newPlayer) {
+        if (leagueType.equals("soccer")) {
```

Comment:
When placing anything other than soccer in the URL where the {leagueType} goes it returns baseball values, I would recommend changing the check for both specific cases and returning a 404 in case the entered value does not exist.

```suggestion
               if (leagueType.equals("soccer")) {
            Teams teamSoccer = soccerLeagues.saveTeams(newTeam);
            log.info("Team created");
            return ResponseEntity.ok(teamSoccer);
        } else if (leagueType.equals("baseball")){
            Teams teamBase = baseBallLeague.saveTeams(newTeam);
            return ResponseEntity.ok(teamBase);
        } else {
            return ResponseEntity.notFound().build();
        }
```